lib/vue/src/components/map/CMap.vue

```diff
@@ -0,0 +1,307 @@
+      @click="onMapClick"
+      @locationfound="onLocationFound"
+      @update:zoom="onZoom"
+      @update:center="onCenter"
+      @update:bounds="onBoundsUpdate"
+    >
+      <l-tile-layer
+        url="https://{s}.tile.openstreetmap.org/{z}/{x}/{y}.png"
+        :attribution="mapOptions.attribution"
+      />
+    
+      <l-polygon
+        v-for="(geometry, i) in polygons"
+        :key="i"
+        :lat-lngs="geometry.map(value => value.geometry)"
+        :color="geometry[0].color"
+      />
+    
+      <l-marker
+        v-for="(marker, i) in markers"
+        :key="`marker-${i}`"
+        :lat-lng="marker"
+        :icon="getIcon(marker.color)"
+        :opacity="activeMarkerIndex === undefined || i == activeMarkerIndex ? 1.0 : 0.6"
+        @click="onMarkerClick(i, marker)"
+      />
+    
+      <l-control class="leaflet-bar">
+        <a
+          v-if="!hideCurrentLocationButton"
+          :title="labels.tooltip && labels.tooltip.goToCurrentLocation"
+          role="button"
+          class="d-flex justify-content-center align-items-center"
+          @click="goToCurrentLocation"
+        >
+          <font-awesome-icon
+            :icon="['fas', 'location-arrow']"
+            class="text-primary"
+          />
+        </a>
+      </l-control>
+    </l-map>
+  </div>
+</template>
+
+<script>
+import { OpenStreetMapProvider } from 'leaflet-geosearch'
+import { divIcon } from 'leaflet'
+import { LControl, LPolygon } from 'vue2-leaflet'
+import CInputSearch from '../input/CInputSearch.vue'
+
+export default {
+  props: {
+    hideCurrentLocationButton: {
+      type: Boolean,
+      default: false
+    },
+    
+    labels: {
+      type: Object,
+      required: true
+    },
+    
+    map: {
+      type: Object
+    },
+    
+    markers: {
+      type: Array,
+      default: () => ([])
+    },
+
+    activeMarkerIndex:{
+      type: Number,
+      default: undefined
+    },
+
+    mapStyle: {
+      type: String,
+      default: "height: 75vh; width: 100%;"
+    },
+
+    polygons: {
+      type: Array,
+      default: () => ([])
+    },
+
+    hideGeoSearch: {
+      type: Boolean,
+      default: false
+    },
+
+    disable: {
+      type: Boolean,
+      default: false
+    }
+  },
+
+  components: {
+    LControl,
+    CInputSearch,
+    LPolygon
+  },
+  
+  data () {
+    return {
+      geoSearch: {
+        query: '',
+        provider: new OpenStreetMapProvider(),
+        results: [],
+      },
+    }
+  },
+
+  computed: {
+    mapOptions () {
+      const defaultOptions = {
+        show: true,
+        zoom: 3,
+        center: [30, 30],
+        rotation: 0,
+        attribution: '&copy; <a target="_blank" href="http://osm.org/copyright">OpenStreetMap</a>',
+      }
+
+      return {
+        ...defaultOptions,
+        ...this.map
+      }
+    }
+  },
+
+  watch: {
+    mapOptions: {
+      handler (map) {
+        if (map.show && this.$refs.map) {
+          this.$refs.map.mapObject.invalidateSize()
```

Comment:
Make sure that map fits block/modal on render. 
If setTimeout is not needed even better